Source/WebCore/rendering/RenderReplaced.h

```diff
@@ -74,10 +74,11 @@ class RenderReplaced : public RenderBox {
 
     void willBeDestroyed() override;
 
+    virtual RenderBox* embeddedContentBox() const { return 0; }
```

Comment:
I'm not a huge fan of way that this bug is fixed by moving which class implements `embeddedContentBox()`. I'd prefer to see an explicit check for "this is an iframe, not an embed" or something.

Reply:
@smfr - Just to understand better, it is to create new function which specifically deals with 'iframe' and then leverage it? Also, can we do this as follow-up since it is interop bug and fails 300 WPT tests?

Reply:
I think Simon is suggesting that you leave the method on `RenderWidget` and `RenderReplaced` untouched, then add a type check in `RenderWidget::embeddedContentBox()`.

So something like:

```
RenderBox* RenderWidget::embeddedContentBox() const
{
    if (!is<RenderEmbeddedObject>(this) || !is<LocalFrameView>(widget()))
        return nullptr;
    return downcast<LocalFrameView>(*widget()).embeddedContentBox();
}
```

Reply:
Above fixes this as well (checked locally) and I also manage to not do any changes in `RenderReplaced.h`.